gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BomConflictCheckTask.java

```diff
@@ -96,7 +97,15 @@ public final void checkBomConflict() {
                             .filter(artifactName -> !recommendationConflicts.contains(artifactName))
                             .map(artifactName -> Pair.of(artifactName, propName));
                 })
-                .collect(Collectors.toMap(Pair::getLeft, Pair::getRight));
+                .collect(Collectors.toMap(
+                        Pair::getLeft,
+                        Pair::getRight,
+                        // Resolve conflicts by choosing the entry that exactly matches the artifact
+                        (propName1, propName2) -> Stream.of(propName1, propName2)
+                                .filter(propName -> !propName.contains("\\*"))
```

Comment:
Talking with @JacekLach proposed an elegant solution to just pick the longest one since it is the most specialized:
We could just replace this by 
```
                .map(propName -> Pair.of(propName, propName.length)))
                .max(Comparator.comparing(Pair::getRight))
                .findFirst().get()
```

Reply:
Actually, regarding @qinfchen comments below. We probably should sort every conflict for the artifact and pick the max if it's a proper max (no other prop of the same length/as specialized), fail otherwise.